From 87f6ea0874c548f15a528ad44347da2e09b8a9d2 Mon Sep 17 00:00:00 2001
From: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
Date: Fri, 16 Dec 2016 14:43:56 +0000
Subject: [PATCH 3/3] ACPI: Drop misplaced acpi_dma_deconfigure() call from
 acpi_bind_one()

Git-commit: 709f94ff018a4403d0bb32643254058d5d9cfa24
Patch-mainline: v4.10-rc3
References: fate#320512, fate#322061

The acpi_bind_one() error return path can be hit either on physical node
allocation failure or if the device being configured is already
associated with an ACPI node and its ACPI companion does not match the
one acpi_bind_one() is setting it up with. In both cases the error
return path is executed before DMA is configured for a device therefore
there is no need to call acpi_dma_deconfigure() on the function error
return path.

Furthermore, if acpi_bind_one() does configure DMA for a device (ie it
successfully executes acpi_dma_configure()) acpi_bind_one() always
completes execution successfully hence there is no need to add an exit
path to deconfigure the DMA set-up (ie by calling acpi_dma_deconfigure()).

Remove the misplaced acpi_dma_deconfigure() in acpi_bind_one() to
reinstate its correct error return path behaviour.

Fixes: d760a1baf20e (ACPI: Implement acpi_dma_configure)
Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/acpi/glue.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/acpi/glue.c b/drivers/acpi/glue.c
index f8d65647ea79..eb709e7c9ffd 100644
--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -250,7 +250,6 @@ int acpi_bind_one(struct device *dev, struct acpi_device *acpi_dev)
 	return 0;
 
  err:
-	acpi_dma_deconfigure(dev);
 	ACPI_COMPANION_SET(dev, NULL);
 	put_device(dev);
 	put_device(&acpi_dev->dev);
-- 
2.11.0

